fetch.go

```diff
@@ -0,0 +1,140 @@
+package namesys
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	ggio "github.com/gogo/protobuf/io"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+
+	pb "github.com/libp2p/go-libp2p-pubsub-router/pb"
+)
+
+const FetchProtoID = protocol.ID("/libp2p/fetch/0.0.1")
```

Comment:
I guess I figured that it would just fetch anything (i.e. `[]byte`), and if for some reason someone wanted to configure their nodes to accept `/ipfs/bafyABC` via fetch and return a block that's something that would be up to them to do. This would also make it easier for people to reuse this protocol within or on top of other protocols.

Is that too generic/confusing for people? Should we just restrict it to records (i.e. things with validators)?

Reply:
I actually really like this. +1